src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -826,7 +827,10 @@ fun processMacroCallVariantsInScope(context: PsiElement, processor: RsResolvePro
     val result = MacroResolver.processMacrosInLexicalOrderUpward(context, processor)
     if (result) return true
 
-    val prelude = context.contextOrSelf<RsElement>()?.findDependencyCrateRoot(STD) ?: return false
+    val element = context.contextOrSelf<RsElement>() ?: return false
+    if (element.crateRoot?.stdlibAttributes == NO_STD) return false
```

Comment:
What's about macros from `core` crate of stdlib?

Reply:
Right, the code didn't handle `core` prelude before, so I thought that it has no macros, but they are actually all re-exported from `std`. Now if `#![no_std]` is present, `core` prelude should be looked up instead.